### PR TITLE
Template lite user registration configurations

### PR DESCRIPTION
**Pull request by DMHP**
Fix https://github.com/wso2/product-is/issues/10896

To enable the lite user registration via configurations, add the following config to the deployement.toml
```
[identity_mgt]
lite_user_registration.enable=true
```

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 1]
```diff
@@ -280,6 +280,14 @@
   "identity_mgt.resend_notification.resend_code_validity": "1m",
   "identity_mgt.password_reset_email.notify_on_reset":false,
 
+  "identity_mgt.lite_user_registration.enable": true,
```
Comment (IsuraD) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:283]: Make this false by default

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -1120,6 +1120,22 @@
         <ResendConfirmationReCaptcha>{{identity_mgt.user_self_registration.enable_resend_confirmation_recaptcha}}</ResendConfirmationReCaptcha>
     </SelfRegistration>
 
+    <LiteRegistration>
+            <Enable>{{identity_mgt.lite_user_registration.enable}}</Enable>
+            <LockOnCreation>{{identity_mgt.lite_user_registration.lock_on_creation}}</LockOnCreation>
+            <Notification>
+                <InternallyManage>{{identity_mgt.lite_user_registration.notification.manage_internally}}</InternallyManage>
+            </Notification>
+            <ReCaptcha>{{identity_mgt.lite_user_registration.enable_recaptcha.enable}}</ReCaptcha>
```
Comment (ayshsandu) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1129]: identity_mgt.lite_user_registration.enable_recaptcha.enable -> enable is repeated.
Change to identity_mgt.lite_user_registration.recaptcha.enable

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 3]
```diff
@@ -280,6 +280,14 @@
...
+  "identity_mgt.lite_user_registration.lock_on_creation": false,
+  "identity_mgt.lite_user_registration.notification.manage_internally": true,
+  "identity_mgt.lite_user_registration.enable_recaptcha.enable": true,
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:286]:
```suggestion
  "identity_mgt.lite_user_registration.recaptcha.enable": true,
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 4]
```diff
@@ -280,6 +280,14 @@
   "identity_mgt.resend_notification.resend_code_validity": "1m",
   "identity_mgt.password_reset_email.notify_on_reset":false,
 
+  "identity_mgt.lite_user_registration.enable": true,
+  "identity_mgt.lite_user_registration.lock_on_creation": false,
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:284]:
```suggestion
  "identity_mgt.lite_user_registration.lock_on_creation": true,
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json (hunk 1 above)
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:283]:
```suggestion
  "identity_mgt.lite_user_registration.enable": false,
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 2 above)
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1129]:
```suggestion
            <ReCaptcha>{{identity_mgt.lite_user_registration.recaptcha.enable}}</ReCaptcha>
```

Comment (AnuradhaSK) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: fixed in https://github.com/wso2/carbon-identity-framework/pull/3315/commits/531de2c815034e255c9dea99628c77c5d83c54a3

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json (hunk 1 above)
Comment (AnuradhaSK) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]: fixed in c0a1eff

**Review by inthirakumaaran — APPROVED** [review]

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 5]
```diff
@@ -280,6 +280,14 @@
   "identity_mgt.resend_notification.resend_code_validity": "1m",
   "identity_mgt.password_reset_email.notify_on_reset":false,
 
+  "identity_mgt.lite_user_registration.enable": false,
+  "identity_mgt.lite_user_registration.lock_on_creation": true,
+  "identity_mgt.lite_user_registration.notification.manage_internally": true,
+  "identity_mgt.lite_user_registration.recaptcha.enable": true,
+  "identity_mgt.lite_user_registration.verification_email_validity": "$ref{identity_mgt.default_mail_validity_period}",
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:287]:
```suggestion
  "identity_mgt.lite_user_registration.verification_email_validity": "1440",
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 6]
```diff
@@ -280,6 +280,14 @@
...
+  "identity_mgt.lite_user_registration.verification_sms_validity": "$ref{identity_mgt.password_reset_sms.sms_otp_validity}",
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:288]:
```suggestion
  "identity_mgt.lite_user_registration.verification_sms_validity": "1",
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 7]
```diff
@@ -280,6 +280,14 @@
...
+  "identity_mgt.lite_user_registration.callback_url": "[${carbon.protocol}://${carbon.host}:${carbon.management.port}].*[/authenticationendpoint/login.do]",
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:289]:
```suggestion
  "identity_mgt.lite_user_registration.callback_url": ".*",
```

**Review by IsuraD — APPROVED** [review]

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 8]
```diff
@@ -280,6 +280,14 @@
   "identity_mgt.resend_notification.resend_code_validity": "1m",
   "identity_mgt.password_reset_email.notify_on_reset":false,
 
+  "identity_mgt.lite_user_registration.enable": false,
+  "identity_mgt.lite_user_registration.lock_on_creation": true,
+  "identity_mgt.lite_user_registration.notification.manage_internally": true,
+  "identity_mgt.lite_user_registration.recaptcha.enable": true,
+  "identity_mgt.lite_user_registration.verification_email_validity": "1440",
+  "identity_mgt.lite_user_registration.verification_sms_validity": "1",
```
Comment (pulasthi7) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:288]: Need to have a unit config for these two

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 9]
```diff
@@ -280,6 +280,14 @@
   "identity_mgt.resend_notification.resend_code_validity": "1m",
   "identity_mgt.password_reset_email.notify_on_reset":false,
 
+  "identity_mgt.lite_user_registration.enable": false,
+  "identity_mgt.lite_user_registration.lock_on_creation": true,
+  "identity_mgt.lite_user_registration.notification.manage_internally": true,
+  "identity_mgt.lite_user_registration.recaptcha.enable": true,
+  "identity_mgt.lite_user_registration.verification_email_validity": "1440",
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:287]:
```suggestion
  "identity_mgt.lite_user_registration.verification_email_validity": "1440m",
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json (hunk 8 above)
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:288]:
```suggestion
  "identity_mgt.lite_user_registration.verification_sms_validity": "1m",
```

Comment (AnuradhaSK) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]: addressed in 1fb2ac1, and efe41c5

**Review by pulasthi7 — APPROVED** [review]

**Pull request merged**